### ISSUE TITLE
exclude does not work when file contains invalid syntax?

### ISSUE THREAD

**Issue by pawamoy** [python/mypy#14670]
**Bug Report**

I'm having an issue with exclude. Basically, I exclude my `tests/fixtures` folder because it contains code with syntax that is only supported by Python 3.8, 3.9, etc, while my project supports Python 3.7. So the files in `tests/fixtures` contain invalid syntax on Python 3.7.

When I run mypy on Python 3.7, it errors with this:

    tests/fixtures/validation_38.py:1: error: invalid syntax [syntax]
    Found 1 error in 1 file (errors prevented further checking)

I exclude the folder with `exclude = tests/fixtures` in `mypy.ini`.
I tried excluding the file name directly, for example with `exclude = validation_38\.py` but it still fails the same way.

**To Reproduce**

```python
# tests/fixtures/validation_38.py
def full(ctx, a: int, /, b: int, *c: int, d: int, e: int = 0, **f: int):
    pass
```

Checking only the parent folder, I'm able to exclude it:

```console
% python3.7 -m mypy --exclude validation_38.py --ignore-missing-imports tests/fixtures/
Success: no issues found in 8 source files
```

Any other way fails:

```console
% python3.7 -m mypy --exclude validation_38.py --ignore-missing-imports tests          
tests/fixtures/validation_38.py:1: error: invalid syntax  [syntax]
Found 1 error in 1 file (errors prevented further checking)
```

```console
% python3.7 -m mypy --exclude fixtures/validation_38.py --ignore-missing-imports tests
tests/fixtures/validation_38.py:1: error: invalid syntax  [syntax]
Found 1 error in 1 file (errors prevented further checking)
```

```console
% python3.7 -m mypy --exclude '.*/validation_38\.py' --ignore-missing-imports tests
tests/fixtures/validation_38.py:1: error: invalid syntax  [syntax]
Found 1 error in 1 file (errors prevented further checking)
```

**Expected Behavior**

Invalid syntax errors should be ignored if the file in question is excluded.
It seemed to work previously (before 1.0.0?).

**Actual Behavior**

An excluded file with invalid syntax makes mypy terminate with an error.

**Your Environment**

- Python version used: 3.7
- Mypy version used: 1.0.0
- Mypy command-line flags: see above
- Mypy configuration options from `mypy.ini` (and other config files):
```ini
[mypy]
ignore_missing_imports = true
exclude = tests/fixtures
warn_unused_ignores = true
show_error_codes = true
```

Possibly related to https://github.com/python/mypy/issues/9981 and https://github.com/python/mypy/issues/6897.

**Comment by adam-grant-hendry** [comment]
This would be useful as well for `cookiecutter` template files that use `'jinja` syntax in import statements, like:

```python
import {{ cookiecutter.project_slug }}
```
